GatewayProxyFactoryBean for unannotated interface creates gateway with empty string errorChannel

**Affects Version(s):** 5.1.3

## Bug report
It is possible to start a flow from an interface class with `IntegrationFlows.from(Class)`.
That interface must not be annotated with `@MessagingGateway`. However, if I use it that way, the `errorChannel` becomes the default "", which leads to problems with error handling subsequently in `org.springframework.integration.channel.MessagePublishingErrorHandler#resolveErrorChannel`.

If I do annotate the interface with `@MessagingGateway` and supply an error channel, it also wants me to define a request channel, which defeats the purpose of `IntegrationFlows.from(Class)`.
I should note that my interface has a single void method.

```
Exception in thread "pool-1-thread-1" org.springframework.messaging.core.DestinationResolutionException: failed to look up MessageChannel with name '' in the BeanFactory.; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean named '' available
	at org.springframework.integration.support.channel.BeanFactoryChannelResolver.resolveDestination(BeanFactoryChannelResolver.java:117)
	at org.springframework.integration.support.channel.BeanFactoryChannelResolver.resolveDestination(BeanFactoryChannelResolver.java:46)
	at org.springframework.integration.channel.MessagePublishingErrorHandler.resolveErrorChannel(MessagePublishingErrorHandler.java:144)
	at org.springframework.integration.channel.MessagePublishingErrorHandler.handleError(MessagePublishingErrorHandler.java:91)
	at org.springframework.integration.util.ErrorHandlingTaskExecutor.lambda$execute$0(ErrorHandlingTaskExecutor.java:60)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean named '' available
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeanDefinition(DefaultListableBeanFactory.java:775)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getMergedLocalBeanDefinition(AbstractBeanFactory.java:1221)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:294)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:273)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:204)
	at org.springframework.integration.support.channel.BeanFactoryChannelResolver.resolveDestination(BeanFactoryChannelResolver.java:89)
	... 7 more
```
Expected: the errorChannel should be undefined or set to a sensible default value, not to an empty string.

--- a/spring-integration-core/src/main/java/org/springframework/integration/gateway/AnnotationGatewayProxyFactoryBean.java
+++ b/spring-integration-core/src/main/java/org/springframework/integration/gateway/AnnotationGatewayProxyFactoryBean.java
@@ -100,7 +100,9 @@ public class AnnotationGatewayProxyFactoryBean extends GatewayProxyFactoryBean {
 		}
 
 		String errorChannel = beanFactory.resolveEmbeddedValue(this.gatewayAttributes.getString("errorChannel"));
-		setErrorChannelName(errorChannel);
+		if (StringUtils.hasText(errorChannel)) {
+			setErrorChannelName(errorChannel);
+		}
 
 		String asyncExecutor = beanFactory.resolveEmbeddedValue(this.gatewayAttributes.getString("asyncExecutor"));
 		if (asyncExecutor == null || AnnotationConstants.NULL.equals(asyncExecutor)) {
@@ -146,11 +148,15 @@ public class AnnotationGatewayProxyFactoryBean extends GatewayProxyFactoryBean {
 
 		String defaultRequestTimeout =
 				beanFactory.resolveEmbeddedValue(this.gatewayAttributes.getString("defaultRequestTimeout"));
-		setDefaultRequestTimeout(Long.parseLong(defaultRequestTimeout));
+		if (StringUtils.hasText(defaultRequestTimeout)) {
+			setDefaultRequestTimeout(Long.parseLong(defaultRequestTimeout));
+		}
 
 		String defaultReplyTimeout =
 				beanFactory.resolveEmbeddedValue(this.gatewayAttributes.getString("defaultReplyTimeout"));
-		setDefaultReplyTimeout(Long.parseLong(defaultReplyTimeout));
+		if (StringUtils.hasText(defaultReplyTimeout)) {
+			setDefaultReplyTimeout(Long.parseLong(defaultReplyTimeout));
+		}
 
 		super.onInit();
 	}

--- a/spring-integration-core/src/test/java/org/springframework/integration/dsl/flows/IntegrationFlowTests.java
+++ b/spring-integration-core/src/test/java/org/springframework/integration/dsl/flows/IntegrationFlowTests.java
@@ -25,6 +25,7 @@ import java.util.concurrent.Executor;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
 import java.util.function.Function;
 import java.util.function.Supplier;
 
@@ -45,6 +46,7 @@ import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Scope;
+import org.springframework.core.task.TaskExecutor;
 import org.springframework.integration.MessageDispatchingException;
 import org.springframework.integration.annotation.MessageEndpoint;
 import org.springframework.integration.annotation.MessagingGateway;
@@ -493,6 +495,26 @@ public class IntegrationFlowTests {
 				.isNotSameAs(this.flow1WithPrototypeHandlerConsumer.getHandler());
 	}
 
+	@Autowired
+	@Qualifier("globalErrorChannelResolutionFunction")
+	private Consumer<String> globalErrorChannelResolutionGateway;
+
+	@Autowired
+	SubscribableChannel errorChannel;
+
+	@Test
+	public void testGlobalErrorChannelResolutionFlow() throws InterruptedException {
+		CountDownLatch errorMessageLatch = new CountDownLatch(1);
+		MessageHandler errorMessageHandler = m -> errorMessageLatch.countDown();
+		this.errorChannel.subscribe(errorMessageHandler);
+
+		this.globalErrorChannelResolutionGateway.accept("foo");
+
+		assertThat(errorMessageLatch.await(10, TimeUnit.SECONDS)).isTrue();
+
+		this.errorChannel.unsubscribe(errorMessageHandler);
+	}
+
 	@MessagingGateway
 	public interface ControlBusGateway {
 
@@ -880,6 +902,16 @@ public class IntegrationFlowTests {
 		public IntegrationFlow flow2WithPrototypeHandler(
 				@Qualifier("myHandler") AbstractReplyProducingMessageHandler handler) {
 			return f -> f.handle(handler, e -> e.id("flow2WithPrototypeHandlerConsumer"));
+		}
+
+		@Bean
+		public IntegrationFlow globalErrorChannelResolutionFlow(@Qualifier("taskScheduler") TaskExecutor taskExecutor) {
+			return IntegrationFlows.from(Consumer.class, "globalErrorChannelResolutionFunction")
+					.channel(c -> c.executor(taskExecutor))
+					.handle((GenericHandler<?>) (p, h) -> {
+						throw new RuntimeException("intentional");
+					})
+					.get();
 		}
 
 	}

